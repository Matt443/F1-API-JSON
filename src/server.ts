import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}