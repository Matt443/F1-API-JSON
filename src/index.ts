import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
