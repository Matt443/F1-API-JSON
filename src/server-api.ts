import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
