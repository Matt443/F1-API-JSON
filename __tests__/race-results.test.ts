import { getRaceResults } from "../src/server";

test("get race results", async () => {
    expect(await getRaceResults(2000)).toMatchSnapshot();
});
