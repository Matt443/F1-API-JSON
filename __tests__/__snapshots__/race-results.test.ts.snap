// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get race results 1`] = `
[
  {
    "car": "Ferrari",
    "date": 2000-03-11T23:00:00.000Z,
    "grandPrix": "Australia",
    "laps": 58,
    "time": "1:34:01.987",
    "winner": "Michael Schumacher",
  },
  {
    "car": "Ferrari",
    "date": 2000-03-25T23:00:00.000Z,
    "grandPrix": "Brazil",
    "laps": 71,
    "time": "1:31:35.271",
    "winner": "Michael Schumacher",
  },
  {
    "car": "Ferrari",
    "date": 2000-04-08T22:00:00.000Z,
    "grandPrix": "San Marino",
    "laps": 62,
    "time": "1:31:39.776",
    "winner": "Michael Schumacher",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-04-22T22:00:00.000Z,
    "grandPrix": "Great Britain",
    "laps": 60,
    "time": "1:28:50.108",
    "winner": "David Coulthard",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-05-06T22:00:00.000Z,
    "grandPrix": "Spain",
    "laps": 65,
    "time": "1:33:55.390",
    "winner": "Mika Hakkinen",
  },
  {
    "car": "Ferrari",
    "date": 2000-05-20T22:00:00.000Z,
    "grandPrix": "Europe",
    "laps": 67,
    "time": "1:42:00.307",
    "winner": "Michael Schumacher",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-06-03T22:00:00.000Z,
    "grandPrix": "Monaco",
    "laps": 78,
    "time": "1:49:28.213",
    "winner": "David Coulthard",
  },
  {
    "car": "Ferrari",
    "date": 2000-06-17T22:00:00.000Z,
    "grandPrix": "Canada",
    "laps": 69,
    "time": "1:41:12.313",
    "winner": "Michael Schumacher",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-07-01T22:00:00.000Z,
    "grandPrix": "France",
    "laps": 72,
    "time": "1:38:05.538",
    "winner": "David Coulthard",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-07-15T22:00:00.000Z,
    "grandPrix": "Austria",
    "laps": 71,
    "time": "1:28:15.818",
    "winner": "Mika Hakkinen",
  },
  {
    "car": "Ferrari",
    "date": 2000-07-29T22:00:00.000Z,
    "grandPrix": "Germany",
    "laps": 45,
    "time": "1:25:34.418",
    "winner": "Rubens Barrichello",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-08-12T22:00:00.000Z,
    "grandPrix": "Hungary",
    "laps": 77,
    "time": "1:45:33.869",
    "winner": "Mika Hakkinen",
  },
  {
    "car": "McLaren Mercedes",
    "date": 2000-08-26T22:00:00.000Z,
    "grandPrix": "Belgium",
    "laps": 44,
    "time": "1:28:14.494",
    "winner": "Mika Hakkinen",
  },
  {
    "car": "Ferrari",
    "date": 2000-09-09T22:00:00.000Z,
    "grandPrix": "Italy",
    "laps": 53,
    "time": "1:27:31.638",
    "winner": "Michael Schumacher",
  },
  {
    "car": "Ferrari",
    "date": 2000-09-23T22:00:00.000Z,
    "grandPrix": "United States",
    "laps": 73,
    "time": "1:36:30.883",
    "winner": "Michael Schumacher",
  },
  {
    "car": "Ferrari",
    "date": 2000-10-07T22:00:00.000Z,
    "grandPrix": "Japan",
    "laps": 53,
    "time": "1:29:53.435",
    "winner": "Michael Schumacher",
  },
  {
    "car": "Ferrari",
    "date": 2000-10-21T22:00:00.000Z,
    "grandPrix": "Malaysia",
    "laps": 56,
    "time": "1:35:54.235",
    "winner": "Michael Schumacher",
  },
]
`;
